### ISSUE TITLE
Segmentation violation in point group code

### ISSUE THREAD

**Issue by dspoel** [openbabel/openbabel#2282]
- [ X] I believe this to be a bug with Open Babel
- [ ] This is a feature request

## Environment Information
Open Babel version: 3.x
Operating system and version: MacOS

## Expected Behavior
Code works

## Actual Behavior
When compiling the attached file as an OB tool and running
```
pgtest water.sdf
```
The code will crash in routines down below the pointgroup code since the atoms do not have a valid parent pointer.

[pointgroup.zip](https://github.com/openbabel/openbabel/files/5286601/pointgroup.zip)

## Steps to Reproduce
See above.

**Comment by dspoel** [comment]
Offending source code:
```
      OBPointGroup obPG;
      obPG.Setup(&mol);
      printf("Point group according to OpenBabel: %s\n", 
             obPG.IdentifyPointGroup());
```

**Comment by ghutchis** [comment]
Yeah, this code needs to be replaced with `libmsym` which is much, much better:
https://github.com/mcodev31/libmsym
